src/pyk/ktool/kprove.py

```diff
@@ -306,10 +305,12 @@ def prove_cterm(
             allow_zero_step=allow_zero_step,
             depth=depth,
         )
-        next_states = list(unique(var_map(ns) for ns in flatten_label('#Or', next_state) if not is_top(ns)))
+        next_states = list(unique(CTerm.from_kast(var_map(ns.kast)) for ns in next_state if not CTerm._is_top(ns.kast)))
```

Comment:
```suggestion
        next_states = list(unique(CSubst(var_map)(ns)) for ns in next_states)
```

I guess we can do something like this instead.

Reply:
Looks good, maybe that we should add this `__call__` method or is `apply` more explicit?

Reply:
(see https://github.com/runtimeverification/pyk/pull/621/commits/450e4e55609e283ed2bff416982e231ef273ca71)

Reply:
Actually it requires us to split the config for the `CSubst` so it would look uglier:

```python
next_states = set()
for ns in next_state:
    config, constraints = split_config_and_constraints(ns.kast)
    next_states.add(CSubst(var_map, flatten_label('#And', constraints)).apply(CTerm.from_kast(config)))
```